projects/core/src/lib/mask.ts

```diff
@@ -149,6 +146,41 @@ export class Maskito extends MaskHistory {
         }
     }
 
+    private updateSelectionRange([from, to]: SelectionRange): void {
+        if (this.element.selectionStart !== from || this.element.selectionEnd !== to) {
+            this.element.setSelectionRange?.(from, to);
+        }
+    }
+
+    private updateValue(value: string): void {
+        this.element.value = value;
```

Comment:
I removed the `if` check because setting the same value does not affect caret position and is negligibly fast (only 10 nanoseconds slower than checking and not doing setting).